src/posit/connect/oauth/oauth.py

```diff
@@ -2,38 +2,49 @@
 
 from typing import Optional, TypedDict
 
-from requests import Session
+from .. import resources
+from .integrations import Integrations
+from .sessions import Sessions
 
-from .config import Config
 
+class OAuth(resources.Resources):
+    def __init__(
+        self, params: resources.ResourceParameters, api_key: str
+    ) -> None:
+        super().__init__(params)
+        self.api_key = api_key
 
-class Credentials(TypedDict, total=False):
-    access_token: str
-    issued_token_type: str
-    token_type: str
-
+    @property
+    def integrations(self):
+        return Integrations(self.params)
 
-class OAuthIntegration:
-    def __init__(self, config: Config, session: Session) -> None:
-        self.url = config.url + "v1/oauth/integrations/credentials"
-        self.config = config
-        self.session = session
+    @property
+    def sessions(self):
+        return Sessions(self.params)
 
     def get_credentials(
         self, user_session_token: Optional[str] = None
     ) -> Credentials:
+        url = self.params.url + "v1/oauth/integrations/credentials"
+
         # craft a basic credential exchange request where the self.config.api_key owner
         # is requesting their own credentials
         data = dict()
         data["grant_type"] = "urn:ietf:params:oauth:grant-type:token-exchange"
         data["subject_token_type"] = "urn:posit:connect:api-key"
-        data["subject_token"] = self.config.api_key
+        data["subject_token"] = self.api_key
```

Comment:
Should we just remove this while we're in here since the `api-key` token exchange has not be implemented yet?

Reply:
yeah, i'm fine with that.